api/envoy/config/core/v3/protocol.proto

```diff
@@ -171,6 +172,13 @@ message AlternateProtocolsCacheOptions {
 
   // Allows pre-populating the cache with entries, as described above.
   repeated AlternateProtocolsCacheEntry prepopulated_entries = 4;
+
+  // Optional list of hostnames suffixes for which Alt-Svc entries can be shared. For example, if
+  // this list contained the value `.c.example.com`, then an Alt-Svc entry for `foo.c.example.com`
+  // could be shared with 'bar.c.example.com' but would not be shared with 'baz.example.com'. On
+  // the other hand, if the list contained the value '.youtube.com' then all three hosts could share
```

Comment:
API looks good, but just a nit on whether we expect to have a leading dot or not for the suffix. Is this a true string suffix or is it a domain zone essentially?